pkg/controller/machinepool/machinepool_controller.go

```diff
@@ -743,23 +753,34 @@ func (r *ReconcileMachinePool) syncMachineSets(
 					}
 				}
 
-				// Carry over the labels on the remote machineset from the generated machineset. Merging strategy preserves the unique labels of remote machineset.
-				if rMS.Spec.Template.Spec.Labels == nil {
-					rMS.Spec.Template.Spec.Labels = make(map[string]string)
-				}
-				// First delete the labels that need to be deleted, then sync the remoteMachineSet labels with that of the generatedMachineSet.
-				for key := range rMS.Spec.Template.Spec.Labels {
-					if labelsToDelete.Has(key) {
-						// Safe to delete while iterating as long as we're deleting the key for this iteration.
-						delete(rMS.Spec.Template.Spec.Labels, key)
-						objectMetaModified = true
```

Comment:
@suhanime I noticed this path used `objectMetaModified` whereas L761 used `objectModified`. They both seem to have the same effect on the outcome, so I consolidated on the latter in my update. Any reason they were different before?

Reply:
I went looking, and the only difference I could see was for [resourcemerge.EnsureObjectMeta](https://github.com/openshift/hive/blob/master/pkg/controller/machinepool/machinepool_controller.go#L710). I don't know if we should be changing that - but I'll leave that upto you.

Reply:
Okay, yeah, it really doesn't look like there's much need for both vars at all, but in any case that the use of the `Meta` one in this context was most likely just a mistake (though a benign one). Let's roll with it :)